craft_parts/errors.py

```diff
@@ -492,21 +492,17 @@ def __init__(self, *, part_name: str) -> None:
         super().__init__(brief=brief)
 
 
-class PluginBuildError(PartsError):
+class UserExecutionError(PartsError):
```

Comment:
This is a superclass for both `PluginBuildError` and `ScriptletRunError` and contains the error processing that was previously in `PluginBuildError`. `ScriptletRunError` is raised when there's an error in scriptlet commands, but if something wrong happens during `craftctl default` execution in the build step `PluginBuildError` will be raised.

Reply:
Why UserExecutionError and not something simpler like BuildStepError?. Is this supposed to be an ABC?

Reply:
Will retain the name per reasons explained by @bepri and change the class to abstract.